pkg/disttask/framework/scheduler/scheduler.go

```diff
@@ -216,15 +218,19 @@ func (s *BaseScheduler) run(ctx context.Context, task *proto.Task) (resErr error
 		if err := s.getError(); err != nil {
 			break
 		}
+		if ctx.Err() != nil {
```

Comment:
`runCtx` is the cancelled ctx on `startCancelCheck`

Reply:
fixed